clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -207,9 +207,10 @@ public synchronized void ensureCoordinatorReady() {
     }
 
     protected RequestFuture<Void> lookupCoordinator() {
-        if (findCoordinatorFuture == null) {
-            findCoordinatorFuture = sendGroupCoordinatorRequest();
-            findCoordinatorFuture.addListener(new RequestFutureListener<Void>() {
+        RequestFuture<Void> future = findCoordinatorFuture;
+        if (future == null) {
+            findCoordinatorFuture = future = sendGroupCoordinatorRequest();
```

Comment:
We need to add some synchronization (double-checked locking or plain synchronization) or CAS here, otherwise we could create two `sendGroupCoordinatorRequest` instances, right?